MDANSE_GUI/Src/MDANSE_GUI/TabbedWindow.py

```diff
@@ -203,6 +232,58 @@ def shut_down(self):
         QApplication.quit()
         self.destroy(True, True)
 
+    @Slot()
+    def populate_recent_trajectory_menu(
+        self, filename=TrajectoryModel.DEFAULT_JSON_PATH
+    ):
+        """Populate the recent trajectory files menu in the File menu."""
+        self.recent_trajectory_fileAct.clear()
+        if os.path.exists(filename):
+            with open(filename) as f:
+                data = json.load(f)
+                data = list(reversed(data))
+                for file in data:
+                    action = QAction(file, self.recent_trajectory_fileAct)
+                    action.triggered.connect(
+                        lambda checked=False, fp=file: self.open_recent_trajectory_file(
+                            fp
+                        )
+                    )
+                    self.recent_trajectory_fileAct.addAction(action)
+        else:
+            return []
```

Comment:
Prefer minimal nesting and indentation


```suggestion
        filepath = Path(filename)

        if not filepath.is_file():
            return []

        with filename.open(encoding="utf-8") as file:
            data = json.load(file)
            for file in data[::-1]:
                action = QAction(file, self.recent_trajectory_fileAct)
                action.triggered.connect(
                    lambda checked=False, fp=file: self.open_recent_trajectory_file(
                        fp
                    )
                )
                self.recent_trajectory_fileAct.addAction(action)
```